auth/simple.go

```diff
@@ -28,30 +28,77 @@ import (
 
 // SimpleAuthConfig holds this configuration necessary for a simple auth implementation.
 type SimpleAuthConfig struct {
+	// AuthN configuration
+	SimpleAuthNConfig SimpleAuthNConfig `yaml:"authN"`
```

Comment:
nit: might be a bit simpler to do:
```
type SimpleAuthConfig struct {
  // Authentication configuration.
  Authentication authenticationConfig `yaml:"authentication"`
  // Authoriztion configuration.
  Authorization authorizationConfig `yaml:"authorization"`
}
```

Reply:
I've changed (agree that it's easier to read and simpler). My intention was to avoid reserving the authenticationConfig, authorizationConfig names for any future implementations in this package.